lib/src/test/java/io/ably/lib/transport/ConnectionManagerTest.java

```diff
@@ -198,4 +198,35 @@ public void connectionmanager_reconnect_default_endpoint() throws AblyException
 		assertThat(connectionManager.getConnectionState().state, is(ConnectionState.connected));
 		assertThat(connectionManager.getHost(), is(equalTo(opts.realtimeHost)));
 	}
+
+	/**
+	 * <p>
+	 * Library and version param 'lib' should include the header value described there
+	 * {@link io.ably.lib.http.HttpUtils#X_ABLY_LIB_VALUE},
+	 * see {@link HttpHeaderTest#header_lib_channel_publish()}
+	 * </p>
+	 * <p>
+	 * Spec: RTN2g
+	 * </p>
+	 */
+	@Test
+	public void connectionmanager_param_lib() throws AblyException {
+		Setup.TestVars testVars = Setup.getTestVars();
+		ClientOptions opts = testVars.createOptions(testVars.keys[0].keyStr);
+		AblyRealtime ably = new AblyRealtime(opts);
+		ConnectionManager connectionManager = ably.connection.connectionManager;
+
+		new Helpers.ConnectionManagerWaiter(connectionManager).waitFor(ConnectionState.connected);
+
+		/* Get X-Ably-Lib param */
+		Param[] params = connectionManager.getConnectParams();
```

Comment:
I am not entirely comfortable that this "really" tests that the connection URL includes the param, it simply tests that `getConnectParams()` returns the expected params.  If for example someone later changed the wS connection mechanic to no longer us this method, the tests would continue to pass even if the lib param may not be used in the URL.  Is there no other way you can think of testing this?  For example, you could run a simple web server that accepts the connection request, it will fail because the upgrade won't happen, but you can inspect the HTTP request to make sure the URL includes the param for example.  WDYT?